lib/records/ccda.rb

```diff
@@ -114,6 +116,36 @@ def self.procedure(procedure, ccda_record)
           "end_time" => time.to_i + 15.minutes,
         })
       end
+
+      def self.careplans(plan, ccda_record)
+        type = plan['type']
+        time = plan['time']
+        entries = [type] + plan['activities']
+        entries.each do |entry| 
+          care_goal = CareGoal.new({
+            'codes' => CAREPLAN_LOOKUP[entry][:codes],
+            'description' => CAREPLAN_LOOKUP[entry][:description],
+            'start_time' => time.to_i,
+            'status' => plan['status'],
+            'reason' => COND_LOOKUP[plan['reason']]
+          })
+          care_goal['end_time'] = plan['stop'].to_i if plan['stop']
+          ccda_record.care_goals << care_goal
+        end
+      end
```

Comment:
I've inspected the resultin C-CDA and noted a few things:

1. Only the narrative is exported.  There is no structured data for the care plan or goals.
2. In the narrative, the plan itself (e.g., _Diabetes self management plan_) is listed as a goal -- so we end up with three goals: the plan itself, diet, and exercise.

If this is a known limitation, fine.  I just wanted to mention it in case it was somehow missed.